src/components/multilineInput/SplitMultilineInput.tsx

```diff
@@ -0,0 +1,167 @@
+                    return (
+                        <Input
+                            classes={styles.input}
+                            value={values[labelId]}
+                            placeholder={input.placeholder}
+                            validate={input.validation ? (value: any) => input.validation(value) : undefined}
+                            key={labelId + inputIndex}
+                            onChange={(value?: string, valid?: boolean) => this.changeValue(value, valid, index, labelId)}
+                        >
+                            <Label>{input.label}</Label>
+                            {deleteButton}
+                        </Input>
+                    );
+                }
+            });
+
+            return (
+                <li key={`split-${index}`} className={styles.inputLine}>
+                    {inputs}
+                </li>
+            );
+        });
+    }
+
+    private getNewInput() {
+        const inputRefs: Input[] = [];
+        const inputs: JSX.Element[] = _.map(this.props.inputs, (input: ISplitInput, inputIndex: number) => {
+            const addButton: JSX.Element = inputIndex + 1 === this.props.inputs.length ? <AddInputAction onClick={() => this.addLine(inputRefs)} /> : null;
+            return (
+                <Input
+                    ref={(ref: Input) => inputRefs.push(ref)}
+                    key={`add-${inputIndex}`}
+                    classes={styles.input}
+                    placeholder={input.placeholder}
+                    validate={input.validation ? (value: any) => input.validation(value) : undefined}
+                >
+                    <Label invalidMessage={input.validationMessage}>{input.label}</Label>
+                    {addButton}
+                </Input>
+            );
+        });
+        return (
+            <li key='new-input' className={styles.inputLine}>
+                {inputs}
+            </li>
+        );
+    }
+
+    private removeLine(index: number) {
+        const values: ISplitValue[] = [
+            ...this.state.values.slice(0, index),
+            ...this.state.values.slice(index + 1),
+        ];
+        this.setState({
+            values,
+        }, this.handleChange);
+    }
+
+    private addLine(inputRefs: Input[]) {
+        let inError: boolean = false;
+        const values: ISplitValue = {};
+        _.each(this.props.inputs, (input: ISplitInput, inputIndex: number) => {
+            const value: any = inputRefs[inputIndex].getInnerValue();
+
+            inputRefs[inputIndex].validate();
+
+            inError = inError || input.validation && !input.validation(value);
+            values[input.id] = value;
+        });
+
+        if (!inError) {
+            this.setState({
+                values: [
+                    ...this.state.values,
+                    values,

```

Comment:
i would have expected a destructuring of `values` here too, maybe rename it to avoid confusion?